### ISSUE TITLE
forward messaging from cargo

### ISSUE THREAD

**Issue by jalil-salame** [axodotdev/cargo-dist#195]
Cargo dist init took 4 minutes 15 seconds to complete

```console
$ cargo dist --version
cargo-dist 0.0.4
```

## Reproduce

I would like to say that you can reproduce it, but it seems to have been a one time thing, any subsecuent tests have yielded times of around ~0.4 seconds.

```console
$ git clone https://github.com/jalil-salame/nsudoku-solve
$ cd nsudoku-solve
$ git checkout 69fbb08
$ rm .github/workflows/release.yml  # Delete previous ci
$ head -n18 Cargo.toml > Cargo.toml.bak # Remove cargo-dist 0.0.2 configuration
$ mv Cargo.toml.bak Cargo.toml
$ time cargo dist init --ci=github
```

**Comment by smallstepman** [comment]
perhaps cargo got stuck doing something else 🤔, but I tried running `cargo build` on some big repo and launching `cargo dist init` and that didn't seem to slow down `cargo dist` too much. There is a chance that `rust-analyzer` got in the way and was trying to reindex your project or whatever, but that wouldn't explain whole 4 minutes (there are too few dependencies in your project to justify 4min-worth of cargo-related work) 

I tried reproducing with 
```bash
echo 'git clone https://github.com/jalil-salame/nsudoku-solve
 cd nsudoku-solve
 git checkout 69fbb08
 rm .github/workflows/release.yml  # Delete previous ci
 head -n18 Cargo.toml > Cargo.toml.bak # Remove cargo-dist 0.0.2 configuration
 mv Cargo.toml.bak Cargo.toml
 time cargo dist init --ci=github && cd .. && rm -rf nsudoku-solve' >> bench.sh
watch -n1 "source bench.sh"
```
and let it run for around two minutes, but it always was between 0.4 and 0.6 sec (that includes the time it takes to `git clone`)

**Comment by Gankra** [comment]
This is almost certainly cargo updating the registry for 4 minutes and us failing to forward the message (as alluded to in #173).

**Comment by ashleygwilliams** [comment]
i agree with @Gankra that it was probably cargo updating- i wonder if it'd be useful to have some reporting to the terminal on what's happening to help make issues like this more obvious- or at least a verbose option that sends the stdout from the wrapped commands through

**Comment by jalil-salame** [comment]
I agree with it being the cargo index, it matches up with me not using `cargo build` for a while in this PC.

It should forward the `cargo` stdout, or at least have a tasks indicator: `[1/4] Running cargo`, so you can debug what is happening.

If nothing else is interesting abot this issue them I am ready for it to be closed.

**Comment by ashleygwilliams** [comment]
closing in favor of #173

**Comment by ashleygwilliams** [comment]
actually, reading more closely #173 is really a diff feature. i'm gonna keep this open as an issue to implement fwding some of the cargo msging

**Comment by Gankra** [comment]
(I couldn't find the reason why stderr wasn't getting forwarded, I feel like I might be missing a flag in guppy or something...)

**Comment by Gankra** [comment]
Ostensibly fixed in 0.0.6 by #207
